packages/core/examples/numericInputExample.tsx

```diff
@@ -0,0 +1,222 @@
+            ], [
+                <label className={Classes.LABEL} key="modifierslabel">Modifiers</label>,
+                this.renderSwitch("Disabled", showDisabled, this.toggleDisabled),
+                this.renderSwitch("Read-only", showReadOnly, this.toggleReadOnly),
+                this.renderSwitch("Left icon", showLeftIcon, this.toggleLeftIcon),
+                this.renderSwitch("Placeholder text", showPlaceholder, this.togglePlaceholder),
+            ],
+        ];
+    }
+
+    protected renderExample() {
+        return (
+            <div>
+                <NumericInput
+                    buttonPosition={BUTTON_POSITIONS[this.state.buttonPositionIndex].value}
+                    intent={this.state.intent}
+
+                    min={MIN_VALUES[this.state.minValueIndex].value}
+                    max={MAX_VALUES[this.state.maxValueIndex].value}
+
+                    stepSize={STEP_SIZES[this.state.stepSizeIndex].value}
+                    majorStepSize={MAJOR_STEP_SIZES[this.state.majorStepSizeIndex].value}
+                    minorStepSize={MINOR_STEP_SIZES[this.state.minorStepSizeIndex].value}
+
+                    disabled={this.state.showDisabled}
+                    readOnly={this.state.showReadOnly}
+                    leftIconName={this.state.showLeftIcon ? SAMPLE_LEFT_ICON : null}
+                    placeholder={this.state.showPlaceholder ? SAMPLE_PLACEHOLDER : null}
+
+                    onUpdate={this.handleUpdate}
+                    value={this.state.value}
+                />
+                <br />
+                <div>
+                    <strong>Value:</strong> <code>{this.state.value || "null"}</code>
+                </div>
+            </div>
+        );
+    }
+
+    private renderSwitch(label: string, checked: boolean, onChange: React.FormEventHandler<HTMLElement>) {
+       return (
+            <Switch
+                checked={checked}
+                label={label}
+                key={this.labelToKey(label)}
+                onChange={onChange}
+            />
+        );
+    }
+
+    private renderSelectMenu(
+        label: string,
+        selectedValue: number | string,
+        options: ISelectOption[],
+        onChange: React.FormEventHandler<HTMLElement>,
+    ) {
+        return (
+            <label className={Classes.LABEL} key={this.labelToKey(label)}>
+                {label}
+                <div className={Classes.SELECT}>
+                    <select value={selectedValue} onChange={onChange}>
+                        {this.renderSelectMenuOptions(options)}
+                    </select>
+                </div>
+            </label>
+        );
+    }
+
+    private renderSelectMenuOptions(options: ISelectOption[]) {
+        return options.map((option, index) => {
+            return <option key={index} value={index}>{option.label}</option>;
+        });
+    }
+
+    private labelToKey(label: string) {
```

Comment:
this doesn't really matter. "My Element Label" is a fine value for a key, React doesn't care. the lowercase convention is more for us lame humans.